### DISCUSSION TITLE
Image coordinate system

### DISCUSSION THREAD

**Discussion by kenavolic** [PlusToolkit/PlusLib#926] [Q&A] [answered]
Hi,

I have calibrated an US probe with "UF" image orientation. Thus, I have an `ImageToProbe` transform corresponding to this orientation (and the depth as well). I then receive the image with same orientation.

I would like to pick points on the image and apply ImageToProbe transform to have their coordinates in the probe reference.

As I see in http://perk-software.cs.queensu.ca/plus/doc/nightly/user/CommonCoordinateSystems.html, the image coordinates represents  the pixel position in an "MF" oriented image. 

Does it mean I have to some extra flip on my picked point before applying `ImageToProbe`?

Thanks a lot,

**Comment by ungi** [comment, marked as answer]
Hi, the image orientation in the device definition is there to make the image coordinate system consistent with the hardware, the physical ultrasound probe. If you check the output of PLUS and the images are not in MF (the coordinates of the first axis do not increase towards the physical mark on the ultrasound probe, etc.) then you can still track your ultrasound correctly. The calibration will compensate for the wrong image orientation definition, and you won't experience any practical errors. But on the long run, it is recommended that you set up image orientation in the device definition correctly. That way your Image coordinate system will be consistent with all other PLUS-based systems.

**Reply by kenavolic** [reply]
Thanks for the answer. Can you just tell me if I am right:

- I have a device with 'UF' orientation
- fcal converts it to 'MF'  and gives me the calibration transform from 'MF' oriented image to probe
- I then use the device with PlusServer with the same device with 'UF' orientation
- My application receives the image with 'MF' orientation (Plus server does the flip as fcal does)
- I pick a point on the image and just have to apply the image2probe transform without any flipping to have the point coordinates in the tracker coordinates

Thanks! I try to understand the process as we have strange emprical results when picking points on the image border while picking points in the center of the image works well (what might look like a flipping issue...)

**Reply by ungi** [reply]
What you write sounds correct. In this case 'UF' just tells PLUS what image orientation to expect from the hardware. Once PLUS gets the image from the hardware, it always transforms every image in 'MF' orientation. That is a good convention to follow, so you always know that every ultrasound data you have is in 'MF' orientation.

**Reply by kenavolic** [reply]
Thanks a lot for your time!
